chajda/tsquery/augments.py

```diff
@@ -53,114 +58,225 @@ def augments_gensim(lang, word, config=Config(), n=5):
+    # Without this, stdout and stderr from fasttext causes doctests to fail.
+    with suppress_stdout_stderr():
+
+        # fasttext downloads models to the current directory,
+        # but we want to store the model in chaja/tsquery/fasttext,
+        # so we change the working directory before downloading
+        try:
+            os.chdir(f"{fasttext_annoy_data_path}/fasttext")
+        except OSError:
+            os.mkdir(f"{fasttext_annoy_data_path}/fasttext")
+            os.chdir(f"{fasttext_annoy_data_path}/fasttext")
+
+        # download and load fasttext model
+        fasttext.util.download_model(lang, if_exists='ignore')
+        fasttext_models[lang] = fasttext.load_model(f"cc.{lang}.300.bin")
+
+        # fasttext downloads two different files for each model,
+        # however, we only need cc.{lang}.300,bin,
+        # so we remove unneeded .gz file if exists
+        if os.path.isfile(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin.gz"):
+            os.remove(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin.gz")
+
+        # change working directory back
+        os.chdir(working_dir)
+
+    # Executing fasttext's get_nearest_neighbor as well as accessing the words in the fasttext model both run
+    # significantly faster after they has been called once,
+    # so we call them once upon loading the model to ensure most efficient runtime for subsequent uses.
+    fasttext_models[lang].get_nearest_neighbors('google', k=5)
+    fasttext_list = []
+    for word in fasttext_models[lang].get_words(on_unicode_error='replace'):
+        fasttext_list.append(word)
+
+
+def destroy_fasttext(lang):
+    '''
+    removes fasttext model from disk and memory
+    '''
+
+    # remove fasttext download from disk
+    if os.path.isfile(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin"):
+        os.remove(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin")
+
+    # remove fasttext model from memory
+    try:
+        del fasttext_models[lang]
+    except KeyError:
+        pass
+
+
+#################################################################################
+# annoy
+#################################################################################
+
+# annoy_indices dictionary will contain languages as keys,
+# and a namedtuple containing the corresponding annoy index, as well as two json files, as values
 annoy_indices = {}
 
+# fasttext_index_for_word dictionary will contain keys for all words in a fasttext language dictionary,
+# and values corresponding to the index of a word;
+# it is populated by the load_annoy_index function,
+# and subsequently written to a json file
+# NOTE:
+# this dictionary is necessary for using annoy for nearest neighbor query after removing fasttext download
+fasttext_index_for_word = {}
 
-def create_annoy_index_if_needed(lang, word='google', n=5, seed=22):
+# fasttext_word_at_index dictionary will contain values for all words in a fasttext language dictionary,
```

Comment:
this comment is the same as the comment above, so don't duplicate the comment; this is one of the reasons why it's good to avoid naming the variables in the comments

Reply:
Does it make more sense to have one comment describing the purpose of both `fasttext_index_for_word` and `fasttext_word_at_index`? For instance, something like:
```
# dictionaries for storing fasttext model words, and their corresponding indices;
# fasttext_index_for_word stores fasttext words as keys, and corresponding indices as values;
# fasttext_word_at_index stores fasttext word indices as keys, and corresponding words as values;
# both dictionaries are populated by the load_annoy_index function,
# and subsequently written to json files
fasttext_index_for_word = {}
fasttext_word_at_index = {}
```
Clearly this does not avoid naming the variables in the comments, however, I feel like it might be helpful in this situation to name drop the dictionaries in the comments as a way of highlighting how the key,value pairs differ.

Another option is just to have one comment that doesn't name the variables in the comments:
```
# dictionaries for storing fasttext model words, and their corresponding indices;
# both dictionaries are populated by the load_annoy_index function,
# and subsequently written to json files
fasttext_index_for_word = {}
fasttext_word_at_index = {}
```

Reply:
either would be fine, but I slightly prefer the top version